lib/run-task.js

```diff
@@ -24,21 +24,23 @@ const spawn = require("./spawn")
 
 const colors = [chalk.cyan, chalk.green, chalk.magenta, chalk.yellow, chalk.red]
 
+let colorIndex = 0
+const taskNamesToColors = {}
+
 /**
  * Select a color from given task name.
  *
  * @param {string} taskName - The task name.
  * @returns {function} A colorize function that provided by `chalk`
  */
 function selectColor(taskName) {
-    let hash = 0
-
-    for (const c of taskName) {
-        hash = ((hash << 5) - hash) + c
-        hash |= 0
+    let color = taskNamesToColors[taskName]
```

Comment:
Thank you for contribution.

The idea sounds good to me.
Could you use `Map` instead of the empty object? I guess that some task names such as `toString` don't work on this implementation.

Reply:
Great point! I guess `npm-run-all` doesn't support Node 0.10 so `Map` is safe to use. I've changed it in the latest commit.